code/datums/components/plumbing/_plumbing.dm

```diff
@@ -234,11 +234,36 @@
 
 ///has one pipe input that only takes, example is manual output pipe
 /datum/component/plumbing/simple_demand
-	demand_connects = NORTH
+	demand_connects = SOUTH
+
+/datum/component/plumbing/simple_demand/hydroponics
+	///Reagent to filter through for reaction purposes, for hydroponics we need water.
+	var/datum/reagent/filtered_chem = /datum/reagent/water
+	///typecast reference to our parent, which should always be a hydroponics tray.
+	var/obj/machinery/hydroponics/constructable/parent_tray
+
+/datum/component/plumbing/simple_demand/hydroponics/Initialize(start, _turn_connects)
+	if(istype(parent, /obj/machinery/hydroponics/constructable))
+		parent_tray = parent
+	. = ..()
+
+/datum/component/plumbing/simple_demand/hydroponics/transfer_to(datum/component/plumbing/target, amount, reagent, datum/ductnet/net)
```

Comment:
This right here is breaking, gone through a few revisions of this now but no avail.